HerPublicWebsite.UnitTests/BusinessLogic/Models/LocalAuthorityDataTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using FluentAssertions;
+using FluentAssertions.Execution;
+using NUnit.Framework;
+using Tests.BusinessLogic.Models.ExpectedLocalAuthorityData;
+using static HerPublicWebsite.BusinessLogic.Models.LocalAuthorityData;
+
+namespace Tests.BusinessLogic.Models;
+
+/*
+ * ⚠ IMPORTANT! ⚠
+ *
+ * The expected data for these tests is deliberately split into multiple files under ExpectedLocalAuthorityData.
+ *
+ * Do NOT refactor these tests to combine these expectations! The separation and duplication is by design.
+ *
+ * These tests were introduced in response to an incident in which some local authorities had their status inadvertently
+ * reverted. This was caused by a mishandled merge that changed the whole LocalAuthorityDetailsByCustodianCode block.
+ *
+ * By separating out the expected data in these tests, we aim to ensure that each piece of information about each local
+ * authority is documented independently. Combining these expectations would risk the exact merging problem that we are
+ * trying to prevent.
+ *
+ * The longer term ideal state would be to move this data into the database, but these tests provide a defensive
+ * mitigation until that is possible.
+ *
+ * For full details, see the report attached to: https://beisdigital.atlassian.net/browse/PC-851
+ */
+[TestFixture]
+public class LocalAuthorityDataTests
+{
+    [Test]
+    public void LocalAuthorityDetailsByCustodianCode_HasCorrectNameForEveryLocalAuthority()
+    {
+        // Arrange
+        var expectedNamesByCode = LocalAuthorityNames.NamesByCustodianCode();
+
+        // Act & Assert
+        const string fieldName = nameof(LocalAuthorityDetails.Name);
+        foreach (var (code, name) in expectedNamesByCode)
+        {
+            using var _ = new AssertionScope($"{fieldName} for LA with code \"{code}\"");
+            var localAuthority = LocalAuthorityDetailsByCustodianCode[code];
+            localAuthority.Name.Should().Be(name);
+        }
+
+        LocalAuthorityDetailsByCustodianCode.Should().HaveSameCount(expectedNamesByCode);
+    }
+
+    [Test]
+    public void LocalAuthorityDetailsByCustodianCode_HasCorrectStatusForEveryLocalAuthority()
+    {
+        // Arrange
+        var expectedStatusesByCode = LocalAuthorityStatuses.StatusesByCustodianCode();
+
+        // Act & Assert
+        const string fieldName = nameof(LocalAuthorityDetails.Status);
+        foreach (var (code, status) in expectedStatusesByCode)
+        {
+            using var _ = new AssertionScope($"{fieldName} for LA with code \"{code}\"");

```

Comment:
These assertion scopes mean that the error messages give you details about where in the {{foreach}} loop the test failed.

For example:
```
Expected Status for LA with code "1150" to be Hug2Status.Pending {value: 1}, but found Hug2Status.Live {value: 2}.
```
rather than just:
```
Expected localAuthority.Status to be Hug2Status.Live {value: 2}, but found Hug2Status.Pending {value: 1}.
```